### DISCUSSION TITLE
Block DNS query to specific IP CIDR but allow every DNS over HTTPS

### DISCUSSION THREAD

**Discussion by matfil92** [AdguardTeam/AdGuardHome#3822] [Q&A]
Hi,
is it possibile to block DNS query with IP from everyone except some whitelist and allow every request through DoH or DoT?

I've tried to put in Settings --> DNS Settings --> Allowed clients only the IP that is OK for me but this block also all request through DoH or DoT.

My scenario is that I've installed ADGuard Home in public VPS, I want restrict DNS access by IP only from my house static IP but I need to leave open for every request by DoT or DoH coming from iPhone iOS profile. 

Please help me
Thanks

**Comment by ainar-g** [comment]
Hello.  There currently isn't any protocol-based filtering in AGH, and plain DNS cannot be disabled.  What you could do is to configure all DoH clients using a single Client ID, for example `everyone` or `public`, and add that to the allowlist.  Would that work for you?

**Comment by matfil92** [comment]
In the allowed list is possibile also to add ClientID and not only CIDR?

**Reply by ainar-g** [reply]
In the beta releases of v0.107, yes.
